enhancements/ingress/wildcard-routing.md

```diff
@@ -0,0 +1,189 @@
+  //
+  // [1] https://github.com/openshift/api/blob/master/route/v1/types.go
+  //
+  // WildcardRouting can not be updated.
+  //
+  // If empty, defaults to "Disabled".
+  //
+  WildcardRouting *WildcardRoutingType `json:"wildcardRouting,omitempty"`
+}
+
+const (
+  // WildcardRoutingDisabled indicates only routes with a wildcard policy
+  // of "None" are admitted by the ingress controller.
+  WildcardRoutingDisabled WildcardRoutingType = "Disabled"
+
+  // WildcardRoutingEnabled indicates routes with any wildcard policy are
+  // admitted by the ingress controller.
+  WildcardRoutingEnabled WildcardRoutingType = "Enabled"
+)
+
+// (Existing type and field, included for completeness)
+type IngressControllerSpec struct {
+  // routeAdmission defines a policy for handling new route claims (for example,
+  // to allow or deny claims across namespaces).
+  //
+  // If empty, defaults will be applied. See specific routeAdmission fields
+  // for details about their defaults.
+  //
+  // +optional
+  RouteAdmission *RouteAdmissionPolicy `json:"routeAdmission,omitempty"`
+}
+
+// (Existing type and field, included for completeness)
+type IngressControllerStatus struct {
+  // routeAdmission is the route admission policy that is in effect.
+  RouteAdmission *RouteAdmissionPolicy
+}
+```
+
+### Migration of existing clusters
+
+All `IngressControllers` with an empty `WildcardRouting` can continue running with the current default, without setting
+a value.
+
+#### Rollbacks
+
+In case of rollbacks where WildcardRoutingEnabled has been used, routes claimed by the ingress controller with
+`wildcardPolicy: Subdomain` must be deleted and created with `wildcardPolicy: None`.
+
+### Test Plan
+
+Test Case 1:
+- Create the default ingress controller.
+- Create two routes, each with a different wildcard policy and ensure that only the route with
+`wildcardPolicy: Subdomain` is admitted.
+
+Test Case 2:
+- Create an ingress controller with `wildcardRouting: Enabled`
+- Create two routes, each with a different wildcard policy and ensure that both are admitted.
+
+Test Case 3:
+- Create an ingress controller with `wildcardRouting: Disabled`
+- Create two routes, each with a different wildcard policy and ensure that only the route with
+`wildcardPolicy: None` is admitted.
+
+Test Case 4:
+- Create an ingress controller with `wildcardRouting: ""`
+- Ensure that the ingress controller is not admitted.
+
+#### Migration
+
+No migration should be required.
+
+### Graduation Criteria
+
+### Upgrade / Downgrade Strategy
+
+See the [rollback section][#rollbacks]
+
+### Version Skew Strategy
+
+The API defines `empty` equivalent to `Disabled`, therefore all the clients will have to keep that check to remain
+backwards compatible.
+
+## Implementation History
+
+## Drawbacks
+
+This only implements one existing feature in the OpenShift Router that existed in OpenShift 3.x, whereas there are many
+more configurations.
```

Comment:
How about replacing this paragraph and the following one with a cross-reference to the "Drawbacks" section of the `route-admission-policy` enhancement?

Reply:
done